### PR TITLE
Fixed galactic coord problem

### LINKED ISSUE #2147: Issue with spatial constraints in Alma.query
The following script:

```python
import matplotlib.pyplot as plt
from astroquery.alma import Alma
from astropy.coordinates import SkyCoord
archive = Alma.query({'spatial_resolution': '0..0.5', 'galactic':'0..360 -2..2'})
coords = SkyCoord(archive['gal_longitude'], archive['gal_latitude'], frame='galactic')
plt.scatter(coords.l.deg, coords.b.deg)
plt.savefig('distribution.png')
```

produces the following plot:

distribution

I would have expected all sources to lie between -2 and 2 in latitude, so it seems there is a bug somewhere along the way?

### PR DESCRIPTION
**Pull request by andamian**
fixes #2147 

Does this look better?
distribution

### CODE REVIEW

**Comment by keflavich** [comment]
Could you repost the original query here?  I think this looks better.

**Comment by andamian** [comment]
before

```select * from ivoa.obscore WHERE (0.0<=spatial_resolution AND spatial_resolution<=0.5) AND (INTERSECTS(RANGE_S2D(268.37542277754466,264.47377034405804,-29.96387952993674,-27.88035232035121), s_region) = 1) AND science_observation='T' AND data_rights='Public'```

now

```select * from ivoa.obscore WHERE (0.0<=spatial_resolution AND spatial_resolution<=0.5) AND (INTERSECTS(RANGE_S2D(264.47377034405804,268.37542277754466,-29.96387952993674,-27.88035232035121), s_region) = 1) AND science_observation='T' AND data_rights='Public'```

**Comment by andamian** [comment]
The unit test fails now because `SkyCoord` automatically makes ra=360 into 0 and that messes up the min/max in formula. Is there a way to prevent that?

**Comment by keflavich** [comment]
SkyCoord has a `wrap_at` attribute; set that to 180deg  - e.g. `coord.wrap_at(180*u.deg)`.

Yes, this is acceptable, but we should note the limitations, and particularly that searches aligned with the Galactic plane are not supported.  I think that's the right interpretation of what you just said

**Comment by andamian** [comment]
@keflavich - the crude translation between galactic range to ICRS range is a crude approximation that probably includes extra areas. The accuracy depends on the size of the range and position. It is however better than the bug that we have right now.

A better alternative would be to approximate the galactic range with a polygon and use that in the ICRS intersect. To do that, we'd need to add more points to the polygon along the latitude parallels to keep the polygon vertices follow the parallels (and not on the spherical arc). More points means more accurate but also more complexity in executing the query. Is this worth from a practical point of view? Are queries like the one in the ticket common enough to justify the effort or the current approximation is sufficient?

**Comment by keflavich** [comment]
Let's use this workaround for now.  I think the polygon approach is much better, but I'm not convinced demand is high enough right now to justify the extra work.

**Comment by andamian** [comment]
Just an update here. I've tested the polygon approach with all transformation on the client side and it seems to work for the most part. An option to do the geometry transformation between different reference systems on the service side with a query function has been just endorsed by IVOA and we might follow that lead for this issue.

**Comment by andamian** [comment]
Fixed

I think this will do it. @keflavich - please review.

File: astroquery/alma/tapsql.py [hunk 1]
```diff
@@ -69,16 +69,21 @@ def _gen_pos_sql(field, value):
...
+                min_pt = coord.SkyCoord(ra_min, dec_min, unit=u.deg)
+                ra_min, dec_min = min_pt.ra.value, min_pt.dec.value
+                max_pt = coord.SkyCoord(ra_max, dec_max, unit=u.deg)
+                ra_max, dec_max = max_pt.ra.value, max_pt.dec.value
+                if frame == 'galactic':
+                    # intersect with s_region is too complicated. ALMA indicated that
+                    # the use of gal_longitude and gal_latitude is good enough
+                    # approximation in this less comon use case
+                    result += ('gal_longitude>={} AND gal_longitude<={} AND '
+                               'gal_latitude>={} and gal_latitude<={}').format(
```
Comment (keflavich) [inline astroquery/alma/tapsql.py:81]: minor typo in the comment.  Is the query case-sensitive?  the third `and` is lower-case.
```suggestion
                    # approximation in this less common use case
                    result += ('gal_longitude>={} AND gal_longitude<={} AND '
                               'gal_latitude>={} and gal_latitude<={}').format(
```

**Review by keflavich — APPROVED** [review]
lgtm, but please verify that the query syntax (lower case `and`) is correct.

**Review by bsipocz — COMMENTED** [review]
This was user reported, so I'm afraid will need a changelog entry. If we merge without one, just leave the label off and I'll deal with it at release time.

**Comment by andamian** [comment]
> lgtm, but please verify that the query syntax (lower case `and`) is correct.

It's case insensitive but I've changed it for consistency. Also, switched to using `coord.Angle` to avoid the wrap up at 360. It is also more appropriate for its purpose (degree representation).

**Comment by andamian** [comment]
> This was user reported, so I'm afraid will need a changelog entry. If we merge without one, just leave the label off and I'll deal with it at release time.

My bad. I wrongly assumed that it's not required because there are no API changes.

**Review by bsipocz — APPROVED** [review]
This is good to go, thanks!

**Pull request merged**